OpenSim/Simulation/Wrap/WrapTorus.cpp

```diff
@@ -137,52 +100,41 @@ void WrapTorus::extendScale(const SimTK::State& s, const ScaleSet& scaleSet)
 
     const double averageXYScale =
         (localScaleVector[0].norm() + localScaleVector[1].norm()) * 0.5;
-   _innerRadius *= averageXYScale;
-   _outerRadius *= averageXYScale;
+   upd_inner_radius() *= averageXYScale;
+   upd_outer_radius() *= averageXYScale;
 }
 
 //_____________________________________________________________________________
 /**
  * Perform some set up functions that happen after the
  * object has been deserialized or copied.
  *
- * @param aModel pointer to OpenSim Model 
+ * @param aModel pointer to OpenSim Model
  */
-void WrapTorus::connectToModelAndBody(Model& aModel, PhysicalFrame& aBody)
+void WrapTorus::extendFinalizeFromProperties()
 {
     // Base class
-    Super::connectToModelAndBody(aModel, aBody);
-
-   // maybe set a parent pointer, _body = aBody;
-
-    if (_innerRadius < 0.0)
-    {
-        string errorMessage = "Error: inner_radius for WrapTorus " + getName() + " was either not specified, or is negative.";
-        throw Exception(errorMessage);
-    }
+    Super::extendFinalizeFromProperties();
+
+    // maybe set a parent pointer, _body = aBody;
+    OPENSIM_THROW_IF_FRMOBJ(
+        get_inner_radius() < 0,
+        InvalidPropertyValue,
+        getProperty_inner_radius().getName(),
+        "Inner radius cannot be less than zero");
+
+    // maybe set a parent pointer, _body = aBody;
+    OPENSIM_THROW_IF_FRMOBJ(
+        get_outer_radius() < 0,
+        InvalidPropertyValue,
+        getProperty_outer_radius().getName(),
+        "Outer Radius cannot be less than zero");
 
-    if (_outerRadius <= _innerRadius)
-    {
-        string errorMessage = "Error: outer_radius for WrapTorus " + getName() + " is less than or equal to inner_radius.";
-        throw Exception(errorMessage);
-    }
```

Comment:
It looks like you removed this check. Any reason? Seems to me that we should retain this check.